### PR TITLE
Add Rector Definitions

### LINKED ISSUE #383: Add definition to each Rector, so it's easy to know what they do without reading whole class
Same as `DefinedFixer` in PHP CS Fixer. Probably make it part of `RectorInterface`, because having 2 paths doesn't make sense.

Suggested by @mssimi, thanks :+1:

### PR DESCRIPTION
**Pull request by TomasVotruba**
Closes #383 

### Followup

- `describe --config / --level ` command to show all details for loaded Rectors, inspire at php-cs-fixer describe command

### Related Changes

- dropped ~4 Rectors that have very rare case of use or are not even needed (no BC break)

### CODE REVIEW

**Comment by carusogabriel** [comment]
This one would only makes sense for me if it is shown somewhere, like README.md or Documentation

**Comment by mssimi** [comment]
We could maybe make command, that shows definition.

**Comment by TomasVotruba** [comment]
I got very similar idea, see the description.

Could you prepare such command with CLi output as default.
And with `--format` options with `markdown` as value?

All you need to add is `getDefinition()` to one Rector and test, if he has this method. Test doesn't have to pass. Just so we have prototype to chat over.

**Comment by TomasVotruba** [comment]
@mssimi @carusogabriel I've created issue for the command https://github.com/rectorphp/rector/issues/408

**Comment by TomasVotruba** [comment]
Tests fail on unrelated issue with Sympliy\BetterReflectoinDocBlock

**Comment by carusogabriel** [comment]
I’ll review this by night, please, do not merge before it :)

**Comment by carusogabriel** [comment]
@TomasVotruba Can you please add what Rectors were dropped in the PR description? Will be easier to track it :+1:

File: src/Rector/AbstractRector.php [hunk 1]
```diff
@@ -94,9 +93,7 @@ public function leaveNode(Node $node)
     public function afterTraverse(array $nodes): array
     {
         $nodes = $this->expressionAdder->addExpressionsToNodes($nodes);
-
         $nodes = $this->propertyAdder->addPropertiesToNodes($nodes);
-
         return $nodes;
```
[the remark this answers is not in the document]
Comment (TomasVotruba) [reply src/Rector/AbstractRector.php]: No extra reason, probably just visual noise. It's not part of PSR-2

File: src/Rector/Architecture/RepositoryAsService/ReplaceParentRepositoryCallsByRepositoryPropertyRector.php [hunk 2]
```diff
@@ -52,7 +54,46 @@ public function refactor(Node $methodCallNode): ?Node
...
+    }
+}
+SAMPLE_TWO
+                ),
+            ]
+        );
+    }
+
     /**
+     * @todo should be part of reflection
```
[the remark this answers is not in the document]
Comment (TomasVotruba) [reply src/Rector/Architecture/RepositoryAsService/ReplaceParentRepositoryCallsByRepositoryPropertyRector.php]: That should be resolve in standalone PR.
It's doesn't have a big priority, just to collect those and resolve when there are more

**Comment by TomasVotruba** [comment]
@carusogabriel I'll get to it. Thanks for the review

**Comment by TomasVotruba** [comment]
@carusogabriel Thank you, I took me a lot time and it will give more clarity to using Rectors.

> A general comment about PHPUnit Rectors. Maybe we can omit TestCase and, what do you think?

Well, first I thought its good idea, to make it shorter. But then I realized that using "in PHPUnit" only, it can be understood as this is refactoring of PHPUnit itself. Which is not the case.

Also there can be other parts of PHPUnit like listeners, test suites, asserts, namespaces etc., so it seem propper here to meain the "TestCase" scope. I just removed the space, to make it more clear it's related to the `TestCase` class.

**Pull request merged**